node/src/protocol/triple.rs

```diff
@@ -0,0 +1,155 @@
+use super::message::TripleMessage;
+use crate::types::TripleProtocol;
+use crate::util::AffinePointExt;
+use cait_sith::protocol::{Action, Participant};
+use cait_sith::triples::TripleGenerationOutput;
+use k256::Secp256k1;
+use std::collections::btree_map::Entry;
+use std::collections::BTreeMap;
+
+/// Unique number used to identify a specific ongoing triple generation protocol.
+/// Without `TripleId` it would be unclear where to route incoming cait-sith triple generation
+/// messages.
+pub type TripleId = u64;
```

Comment:
Would this id also be used to track the uniqueness of the triple used?

Reply:
Yes, it is not the end of the world if two of them happen to have the same id (although it shouldn't happen, picking two 64-bit numbers that are the same is very inprobable). It will just mean that the triple generation for this specific triple is going to fail and the nodes would have to give up on it.